plugins/azuread/pkg/fetch/fetch.go

```diff
@@ -84,24 +55,55 @@ func (f *Fetcher) Fetch(ctx context.Context, outputWriter, errorWriter io.Writer
 			return err
 		}
 
-		userBytes, err := writer.GetSerializedContent()
-		if err != nil {
-			common.WriteErrorWithExitCode(errorWriter, err, 1)
+		if err := writeObject(jsonWriter, writer, errorWriter); err != nil {
 			return err
 		}
+	}
+
+	return nil
+}
+
+func (f *Fetcher) fetchGroups(ctx context.Context, jsonWriter *js.JSONArrayWriter, errorWriter io.Writer) error {
+	aadGroups, err := f.azureClient.ListGroups(ctx, f.groupProps)
+	if err != nil {
+		common.WriteErrorWithExitCode(errorWriter, err, 1)
+	}
 
-		userString := "{" + string(userBytes) + "}"
+	for _, group := range aadGroups {
+		writer := kiota.NewJsonSerializationWriter()
 
-		var obj map[string]interface{}
-		if err := json.Unmarshal([]byte(userString), &obj); err != nil {
+		err := group.Serialize(writer)
+		if err != nil {
 			common.WriteErrorWithExitCode(errorWriter, err, 1)
 			return err
 		}
 
-		if err := jsonWriter.Write(obj); err != nil {
-			_, _ = errorWriter.Write([]byte(err.Error()))
+		if err := writeObject(jsonWriter, writer, errorWriter); err != nil {
+			return err
 		}
 	}
 
 	return nil
 }
+
+func writeObject(jsonWriter *js.JSONArrayWriter, writer *kiota.JsonSerializationWriter, errorWriter io.Writer) error {
```

Comment:
The signature of this function is extremely confusing! What does it mean for a function to take three different kinds of writers as arguments? Which ones are input? Which are output? What order are they in?
It looks totally random.